Firestore/core/test/unit/model/mutation/overlay_test.cc

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+#include <type_traits>
+#include <utility>
+
+#include "Firestore/core/src/model/mutation.h"
+#include "Firestore/core/src/model/mutation/overlay.h"
+#include "Firestore/core/src/model/patch_mutation.h"
+#include "Firestore/core/src/model/resource_path.h"
+#include "Firestore/core/test/unit/testutil/equals_tester.h"
+#include "Firestore/core/test/unit/testutil/testutil.h"
+#include "absl/strings/string_view.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+namespace mutation {
+namespace {
```

Comment:
I have some general comments, no action needed however:

* There are a lot cool ways I did not know to write tests!
* A big chunk of the tests are actually testing C++ compiler doing the right thing IMO. I am not asking you to remove, it's just, just...I'd not have added these tests myself, and I think that is fine..